### PR TITLE
move to dag_run.logical_date from execution date in OpenLineage provider

### PR DESCRIPTION
**Pull request by mobuchowski**
Remove usage of deprecated `execution_date` in favor of `logical_date`

### CODE REVIEW

File: tests/providers/openlineage/plugins/test_listener.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def mock_task_id(dag_id, task_id, try_number, execution_date):
     task_instance.state = State.RUNNING
     task_instance.start_date = dt.datetime(2023, 1, 1, 13, 1, 1)
     task_instance.end_date = dt.datetime(2023, 1, 3, 13, 1, 1)
-    task_instance.execution_date = "execution_date"
+    task_instance.execution_date = "logical_date"
```
Comment (Avihais12344) [inline tests/providers/openlineage/plugins/test_listener.py]: It should be a certain date, and not just a string, I think. Maybe we can fix it here.

**Comment by potiuk** [comment]
Is it backwards-compatible with Airflow 2?

**Comment by mobuchowski** [comment]
@potiuk `logical_date` property was added back in 2021

Screenshot 2024-09-02 at 11 14 00

**Comment by potiuk** [comment]
> logical_date teproperty was added back in 2021

I know, just checking if it is ok to change it in the provider - which will be (for the next 6 months or so) only installed on Airflow 2 :)

**Comment by potiuk** [comment]
Simply execution_date and logical_date have different semantics, so my question is, if that will impact observed open-lineage events. And if so - whether it should be somehow explained in the changelog so that users are not surprised.

**Comment by mobuchowski** [comment]
@potiuk does it really have different semantics?

From what I can find, it's essentially a rename: 

From AIP-39
> We need a single date value that is the "canonical" date for the DagRun date, that can be used for sorting DagRuns for the UI view, so the current "execution_date" column on DagRun needs to be kept, but we should rename - and we've chosen "logical_date".

Also [in the docstring](https://github.com/apache/airflow/pull/16352/files#diff-ffc28f0462bb68924dcaa55031ad885759562f6cb589744dd1e123dbbcd2ecf6R88):
> This replaces ``execution_date`` in Airflow 2.1 and prior. The idea is essentially the same, just a different name.

Basically, we want the same: the time when the DagRun is scheduled to be run, without consideration if it's actually going to run at that time, due to scheduler being backed up or other reason.

**Comment by potiuk** [comment]
> Basically, we want the same: the time when the DagRun is scheduled to be run, without consideration if it's actually going to run at that time, due to scheduler being backed up or other reason.

Yep. You are right. I had confused it with "start/end interval" and other macros available in context, but logical_date and execution_date are really aliases.

**Review by potiuk — APPROVED** [review]

File: tests/providers/openlineage/plugins/test_listener.py (hunk 1 above)
Comment (mobuchowski) [reply tests/providers/openlineage/plugins/test_listener.py]: Fixed.

**Pull request merged**